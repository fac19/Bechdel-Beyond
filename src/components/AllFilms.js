import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import  Grid  from "@material-ui/core/Grid";
import Card from "@material-ui/core/Card";
import CardMedia from "@material-ui/core/CardMedia";
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

import getData from "../utils/getData";

const useStyles = makeStyles((theme) => ({
  media: {
    height: '300px',
    width: '200px', 
  },
}));

export default function AllFilms() {
  const [filmData, setFilmData] = React.useState('')
  const classes = useStyles()

  React.useEffect(() => {
    getData('/films').then(data => setFilmData(data))}, [])
  
    if (!filmData) return null

  console.log(filmData)

  const showPoster = (poster) => {
    if (poster.status === 404){
     return "https://stockpictures.io/wp-content/uploads/2020/01/image-not-found-big-768x432.png"

    }
    return poster
  }


return (
    <div>
      <Grid container spacing={40} justify="center">
        {filmData.map(film => (
          <Grid item key={film.title}>
            <Card>
              <CardActionArea>
        <CardMedia
          className={classes.media}
          image={showPoster(film.poster)}
          title={film.title}
        />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2">
            {film.title}
          </Typography>
        </CardContent>
      </CardActionArea>
            </Card>
          </Grid>
        ))}
      </Grid>
    </div>
  );
}

