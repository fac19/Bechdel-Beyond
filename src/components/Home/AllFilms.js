import React from 'react';
import { Link } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';

import Paper from '@material-ui/core/Paper';
import InputBase from '@material-ui/core/InputBase';
import SearchIcon from '@material-ui/icons/Search';
import IconButton from '@material-ui/core/IconButton';
import Divider from '@material-ui/core/Divider';

import Grid from '@material-ui/core/Grid';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles((theme) => ({
	card: {
		width: '300px',
		maxHeight: '550px',
		margin: '.7rem',
		display: 'flex',
		flexDirection: 'column',
		justifyContent: 'center',
	},
	media: {
		height: '250px',
	},
	cardlink: {
		textDecoration: 'none',
	},
	title: {
		fontSize: '1.3rem',
		fontWeight: 'bold',
		marginBottom: '5px',
		fontFamily: 'Merriweather, serif',
	},
	subtitle: {
		fontSize: '1rem',
		margin: '3% auto',
		fontFamily: 'Montserrat, sans-serif',
	},
	plot: {
		fontSize: '14px',
		marginBottom: '7px',
		fontFamily: 'Montserrat, sans-serif',
	},
	ratings: {
		fontWeight: 'bold',
	},
	root: {
		padding: '2px 4px',
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
		width: '75%',
	},
	input: {
		marginLeft: theme.spacing(1),
		flex: 1,
		fontFamily: 'Montserrat, sans-serif',
	},
	iconButton: {
		padding: 10,
	},
	searchWrapper: {
		marginBottom: '4%',
		padding: '40px 10px 10px 10px',
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
	},
}));

export default function AllFilms({ filmData }) {
	const [movieTitle, setMovieTitle] = React.useState('');
	const [currentMovies, setCurrentMovies] = React.useState([]);
	const classes = useStyles();
	React.useEffect(() => {
		if (!movieTitle) {
			setCurrentMovies(filmData);
		} else {
			let filteredFilms = filmData.filter((film) => {
				if (film.title.includes(movieTitle)) {
					return film;
				} else {
					setCurrentMovies(null);
					return null;
				}
			});

			setCurrentMovies(filteredFilms);
		}
	}, [movieTitle, filmData]);

	if (!filmData) return null;
	console.log(filmData);
	const showPoster = (poster) => {
		if (poster.status === 404) {
			return 'https://stockpictures.io/wp-content/uploads/2020/01/image-not-found-big-768x432.png';
		}
		return poster;
	};

	return (
		<>
			<div className={classes.searchWrapper}>
				<Paper
					component="form"
					className={classes.root}
					onSubmit={(e) => e.preventDefault()}
				>
					<InputBase
						className={classes.input}
						onChange={(e) => setMovieTitle(e.target.value)}
						placeholder="Search Movies"
						inputProps={{ 'aria-label': 'search movies', 'data-cy': 'search' }}
					/>
					<IconButton
						type="submit"
						className={classes.iconButton}
						aria-label="search"
					>
						<SearchIcon />
					</IconButton>
				</Paper>
			</div>
			<Grid container justify="center">
				{currentMovies.length < 1 ? (
					<Typography data-cy={'not-found'} variant="h5" component="h3">
						No movies from your search
					</Typography>
				) : (
					<>
						{currentMovies.map((film) => (
							<Grid data-cy={'movie-card-container'} item key={film.movapi_id}>
								<Card
									className={classes.cardlink}
									data-cy={'movie-card'}
									component={Link}
									to={`/film/${film.title}`}
								>
									<CardActionArea className={classes.card}>
										<CardMedia
											className={classes.media}
											component={'img'}
											src={showPoster(film.poster)}
											title={film.title}
										/>

										<CardContent gutterbottom="true">
											<Typography
												data-cy={'card-title'}
												className={classes.title}
												variant="h5"
												component="h3"
											>
												{film.title}
											</Typography>
											<Divider className={classes.divider} />
											<Typography className={classes.subtitle}>
												{'Synopsis: '}
											</Typography>
											<Typography
												className={classes.plot}
												variant="body1"
												component="p"
											>
												{film.plot}
											</Typography>
											<div className={classes.ratings}>
												<Typography>Metascore: {film.ratings}</Typography>
												<Typography></Typography>
											</div>
										</CardContent>
									</CardActionArea>
								</Card>
							</Grid>
						))}
					</>
				)}
			</Grid>
		</>
	);
}
