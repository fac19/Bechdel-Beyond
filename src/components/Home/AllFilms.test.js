import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
// import { shallow, mount, render } from 'enzyme';

import AllFilms from './AllFilms';
// import getData from '../../utils/getData';

it('should render <loading> in AllFilms page if no data', () => {
	const filmCards = renderer
		.create(
			<Router>
				<AllFilms />
			</Router>,
		)
		.toJSON();
	expect(filmCards).toMatchSnapshot();
});

it('should render AllFilms page correctly if data provided', () => {
	const filmData = [
		{
			title: 'Titanic',
			movapi_id: '34',
			poster: 'https://m.media-amazon.com/images...',
			year: '1997',
			rated: 'PG-13',
			runtime: '194min',
			genre: "['drama', 'romance']",
			plot: 'A seventeen-year-old aristocrat falls in love with...',
			filmLanguage: 'English',
			country: 'USA',
			awards: 'Won 11 oscars and another 114 wins',
			ratings: '86',
		},
		{
			title: 'Titanic',
			movapi_id: '35',
			poster: 'https://m.media-amazon.com/images...',
			year: '1997',
			rated: 'PG-13',
			runtime: '194min',
			genre: "['drama', 'romance']",
			plot: 'A seventeen-year-old aristocrat falls in love with...',
			filmLanguage: 'English',
			country: 'USA',
			awards: 'Won 11 oscars and another 114 wins',
			ratings: '86',
		},
	];
	const filmCards = renderer
		.create(
			<Router>
				<AllFilms filmData={filmData} />
			</Router>,
		)
		.toJSON();
	expect(filmCards).toMatchSnapshot();
});
