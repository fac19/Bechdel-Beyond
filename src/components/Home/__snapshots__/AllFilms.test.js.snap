// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <loading> in AllFilms page if no data 1`] = `null`;

exports[`should render AllFilms page correctly if data provided 1`] = `
Array [
  <div
    className="makeStyles-searchWrapper-11"
  >
    <form
      className="MuiPaper-root makeStyles-root-8 MuiPaper-elevation1 MuiPaper-rounded"
      onSubmit={[Function]}
    >
      <div
        className="MuiInputBase-root makeStyles-input-9"
        onClick={[Function]}
      >
        <input
          aria-label="search movies"
          className="MuiInputBase-input"
          data-cy="search"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Search Movies"
          type="text"
        />
      </div>
      <button
        aria-label="search"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-10"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
      data-cy="movie-card-container"
    >
      <a
        className="MuiPaper-root MuiCard-root makeStyles-cardlink-3 MuiPaper-elevation1 MuiPaper-rounded"
        href="/film/Titanic"
        onClick={[Function]}
      >
        <button
          className="MuiButtonBase-root MuiCardActionArea-root makeStyles-card-1"
          data-cy="movie-card"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <img
            className="MuiCardMedia-root makeStyles-media-2 MuiCardMedia-media MuiCardMedia-img"
            src="https://m.media-amazon.com/images..."
            title="Titanic"
          />
          <div
            className="MuiCardContent-root"
            gutterbottom="true"
          >
            <h3
              className="MuiTypography-root makeStyles-title-4 MuiTypography-h5"
              data-cy="card-title"
            >
              Titanic
            </h3>
            <hr
              className="MuiDivider-root"
            />
            <p
              className="MuiTypography-root makeStyles-subtitle-5 MuiTypography-body1"
            >
              Synopsis: 
            </p>
            <p
              className="MuiTypography-root makeStyles-plot-6 MuiTypography-body1"
            >
              A seventeen-year-old aristocrat falls in love with...
            </p>
            <div
              className="makeStyles-ratings-7"
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                Metascore: 
                86
              </p>
              <p
                className="MuiTypography-root MuiTypography-body1"
              />
            </div>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight"
          />
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
      data-cy="movie-card-container"
    >
      <a
        className="MuiPaper-root MuiCard-root makeStyles-cardlink-3 MuiPaper-elevation1 MuiPaper-rounded"
        href="/film/Titanic"
        onClick={[Function]}
      >
        <button
          className="MuiButtonBase-root MuiCardActionArea-root makeStyles-card-1"
          data-cy="movie-card"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <img
            className="MuiCardMedia-root makeStyles-media-2 MuiCardMedia-media MuiCardMedia-img"
            src="https://m.media-amazon.com/images..."
            title="Titanic"
          />
          <div
            className="MuiCardContent-root"
            gutterbottom="true"
          >
            <h3
              className="MuiTypography-root makeStyles-title-4 MuiTypography-h5"
              data-cy="card-title"
            >
              Titanic
            </h3>
            <hr
              className="MuiDivider-root"
            />
            <p
              className="MuiTypography-root makeStyles-subtitle-5 MuiTypography-body1"
            >
              Synopsis: 
            </p>
            <p
              className="MuiTypography-root makeStyles-plot-6 MuiTypography-body1"
            >
              A seventeen-year-old aristocrat falls in love with...
            </p>
            <div
              className="makeStyles-ratings-7"
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                Metascore: 
                86
              </p>
              <p
                className="MuiTypography-root MuiTypography-body1"
              />
            </div>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight"
          />
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
  </div>,
]
`;
